spec/std/http/response_spec.cr

```diff
@@ -78,6 +80,49 @@ module HTTP
       response.headers["Content-Length"].should eq("5")
     end
 
+    it "sets content length even without body" do
+      response = Response.new(200)
+      response.headers["Content-Length"].should eq("0")
+    end
+
+    it "doesn't sets content length for 1xx, 204 or 304" do
+      [100, 101, 204, 304].each do |status|
+        response = Response.new(status)
+        response.headers.length.should eq(0)
+      end
+    end
+
+    it "raises when creating 1xx, 204 or 304 with body" do
+      errors = 0
+      [100, 101, 204, 304].each do |status|
+        begin
+          Response.new(status, "hello")
+        rescue ArgumentError
+          errors += 1
+        end
+      end
+      errors.should eq(4)
+    end
+    
+    it "allows reading the body as an IO" do
+      response = Response.from_io(StringIO.new("HTTP/1.1 200 OK\r\nContent-Length: 5\r\n\r\nhelloworld"))
+      response.body?.should be_true
+      response.body_io.read(3).should eq("hel")
+      response.body_io.read.should eq("lo")
+    end
+    
+    it "raises when trying to fetch the body after an IO read" do
+      error = nil
+      begin
+        response = Response.from_io(StringIO.new("HTTP/1.1 200 OK\r\nContent-Length: 5\r\n\r\nhelloworld"))
+        response.body_io.read(3)
+        response.body
+      rescue ex
+        error = ex
+      end
```

Comment:
We got an `expect_raises` macro for these cases, `expect_raises ArgumentError do`